File: app/src/main/java/es/etologic/mahjongscoring2/data/repository/local/converters/DateConverter.java
package es.etologic.mahjongscoring2.data.repository.local.converters;

import android.arch.persistence.room.TypeConverter;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Locale;

public class DateConverter {

    private static SimpleDateFormat englihSimpleDateFormat =
            new SimpleDateFormat("EEEE MM/dd/yyyy hh:mm", Locale.ENGLISH);

    @TypeConverter
    public static String toFormattedDateTime(Long timestamp) {
        return timestamp == null ? null : englihSimpleDateFormat.format(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(String formattedDateTime) throws ParseException {
        return formattedDateTime == null ? null :
                englihSimpleDateFormat.parse(formattedDateTime).getTime();
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/converters/DateConverter.kt
package es.etologic.mahjongscoring2.data.local_data_source.local.converters

import androidx.room.TypeConverter
import java.util.*

internal class DateConverter {
    
    internal companion object {
        
        @TypeConverter
        @JvmStatic
        internal fun toDate(timestamp: Long?): Date? {
            return if (timestamp == null) null else Date(timestamp)
        }
        
        @TypeConverter
        @JvmStatic
        internal fun toTimestamp(date: Date?): Long? {
            return date?.time
        }
    }
}